### PR TITLE
mainpagetemplateinitcommit

### PR DESCRIPTION
**Pull request by Pranaav003**
Mainpage template of NewsWebApp.

### CODE REVIEW

**Comment by zeshengchen** [comment]
If this is your first pull request, please invite me to peer review it as well!

**Review by jude-lieb — APPROVED** [review]
Ok

**Comment by zeshengchen** [comment]
Please add the description for this pull request - what are main changes in this pull request?

File: frontend/originaltemplate.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        </div>
+    </nav>
+
+    <!-- News Grid - Dynamic Section -->
+    <div class="container mx-auto grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 p-10 w-50" id="news-grid">
+        <!-- News articles will be injected here by JavaScript -->
+    </div>
+
+    <!-- Reference to the external JS file -->
+    <script src="./javascript/mainpage.js"></script>
```
Comment (zeshengchen) [inline frontend/originaltemplate.html:76]: It is better to put "script" tags in the "head" section!

Comment (zeshengchen) [reply frontend/originaltemplate.html:76]: If needed, use "onload" for "body" tag to trigger the function call in javascript file after the page is loaded.

**Comment by zeshengchen** [comment]
Not sure why this pull request has been there for a while, but no action to fix the peer review comments?

File: frontend/originaltemplate.html (hunk 1 above)
Comment (Pranaav003) [reply frontend/originaltemplate.html:76]: You're absolutely correct that placing <script> tags in the <head> section is the right convention , but when I do so,  I end up blocking the rendering of the page, which prevents the grid from coming into view.

**Pull request closed without merging**